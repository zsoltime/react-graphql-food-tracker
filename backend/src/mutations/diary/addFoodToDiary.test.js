'use strict';

const addFoodToDiary = require('./addFoodToDiary');

describe('addFoodToDiary mutation', () => {
  it('throws error if food ID is not valid', () => {
    const args = {
      id: 'invalidFoodId',
      servingSize: 77,
      servingUnit: 'G',
    };
    const mockedContext = {
      db: {
        query: {
          food: jest
            .fn()
            .mockImplementation(() => Promise.resolve(null)),
        },
      },
      request: { userId: 'validUserId' },
    };

    expect(
      addFoodToDiary(null, args, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
    expect(mockedContext.db.query.food).toHaveBeenCalled();
    expect(
      mockedContext.db.query.food.mock.calls[0]
    ).toMatchSnapshot();
  });

  it('throws error if user not signed in', () => {
    const mockedContext = {
      request: {},
    };

    expect(
      addFoodToDiary(null, null, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
