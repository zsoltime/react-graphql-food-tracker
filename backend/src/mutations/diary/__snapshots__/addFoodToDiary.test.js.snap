// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addFoodToDiary mutation throws error if food ID is not valid 1`] = `
Array [
  Object {
    "where": Object {
      "id": "invalidFoodId",
    },
  },
]
`;

exports[`addFoodToDiary mutation throws error if food ID is not valid 2`] = `"Food doesn't exist"`;

exports[`addFoodToDiary mutation throws error if user not signed in 1`] = `"You must be signed in to log food"`;
